[Zhang YQ] ip

BigheartSmallbrain opened the pull request:
##### Duke V1.0

>"The moment you give up is the moment you let someone else win"– Kobe Bryant ([source](https://www.inc.com/marcel-schwantes/15-kobe-bryant-quotes-from-his-legendary-career-that-will-inspire-you.html))

DukeV1.0 manages your tasks with simple typing. It's

*text-based
*easy to learn
*~~FAST~~  SUPER FAST to use

All you need to do is,

download it from [here](https://github.com/BigheartSmallbrain/ip/releases/tag/A-Jar).
run duke.jar
add your tasks.
let it manage your tasks for you 😉
And it is **FREE!**

Features:

 - [x] Managing tasks
 - [x] Managing deadlines
 - [ ] Reminders (coming soon)

If you Java programmer, you can use it to practice Java too. Here's the main method:
```java
public class Duke {
  public static void main(String[] args) {
        new Duke("data/tasks.txt").run();
    }
}
```

Commit:
Add Gradle support

BigheartSmallbrain committed:
duke level 1 completed

BigheartSmallbrain committed:
Level 2 completed

BigheartSmallbrain committed:
level2: added features(add and list) to chatbot

BigheartSmallbrain committed:
Level 3 - added new feature of list and mark/unmark tasks in the list

BigheartSmallbrain committed:
Level 3 - added Task class. Added new display for list. Added new feature of mark/unmark tasks in list

BigheartSmallbrain committed:
created new class: Deadline, Event, Todo and updated main in Duke

BigheartSmallbrain committed:
changed input.txt to simulate test automation

BigheartSmallbrain committed:
updated Duke.java with exceptions and created DukeException class

BigheartSmallbrain committed:
added delete feature to duke.java

BigheartSmallbrain committed:
run test and passed

BigheartSmallbrain committed:
Add save to file function

BigheartSmallbrain committed:
Add formatting dates and time feature

BigheartSmallbrain committed:
merge branch-level-7 to master

BigheartSmallbrain committed:
resolve merge conflict when merging branch-level-8 to master

BigheartSmallbrain committed:
Add more classes for OOP

BigheartSmallbrain committed:
Add more classes for OOP

BigheartSmallbrain committed:
package duke , duke.task and duke.command

BigheartSmallbrain committed:
Merge remote-tracking branch 'origin/add-gradle-support'
Merging add-gradle-support to master branch

BigheartSmallbrain committed:
Add gradle to automate build tasks of duke

BigheartSmallbrain committed:
Add JUnit Tests for DeadlineCommand and Event class

BigheartSmallbrain committed:
modify files for JavaDoc

BigheartSmallbrain committed:
Modify for coding standard

BigheartSmallbrain committed:
Add Find feature to Duke

BigheartSmallbrain committed:
Merge branch 'branch-A-CodingStandard'

BigheartSmallbrain committed:
Merge branch 'branch-Level-9'

FireRadical22 commented on src/main/java/duke/Parser.java:
```diff
@@ -0,0 +1,70 @@
+     */
+    public static Command parse(String command) throws DukeException {
+            String[] commandArray = command.split(" ", 2);
+            switch (commandArray[0].toUpperCase()) {
```
Should these 2 lines above have 4 spaces for indentation?

FireRadical22 commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,114 @@
+public class Storage {
+    private final String STORAGE_PATH;
+
+    public Storage(String filePath) {
```
Perhaps you can consider using java.nio.Path to specify file path in an OS-independent way. Additionally, you can maybe consider using java.nio.Files to perform write and read functions.

BigheartSmallbrain replied:
I think this tip is very helpful, would like to tryout this method of implementation.

FireRadical22 commented on src/main/java/duke/TaskList.java:
```diff
@@ -0,0 +1,49 @@
+        this.tasksList.set(num - 1, currTask);
+    }
+
+    public Task get(int num) {
```
```suggestion
    public Task getTask(int num) {
```

FireRadical22 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,38 @@
+    /**
+     * Updates the boolean value of isDone to true
+     */
+    public void check() {
```
```suggestion
    public void checkTask() {
```

FireRadical22 commented on src/main/java/duke/task/Task.java:
```diff
@@ -0,0 +1,38 @@
+    /**
+     * Updates the boolean value of isDone to false
+     */
+    public void uncheck() {
```
```suggestion
    public void uncheckTask() {
```

FireRadical22 commented on src/main/java/duke/Duke.java:
```diff
@@ -0,0 +1,56 @@
+                }
+            }
+        }
+
```
Should the body for run() have an indentation level of 2 tabs instead of 1?

FireRadical22 reviewed (commented):
Overall, the code looks fairly clean. I like the organisation of commands into their respective classes. However, it would be nice to see documentation for all the methods and classes so as to get a better understanding of them. Additionally, there were a few instances where you could have ensure that the indentation level of the line is correct.

BigheartSmallbrain committed:
Add checkstyle feature and modified codebase

BigheartSmallbrain committed:
Add GUI using JavaFX

UnwilledTangent commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,114 @@
+            String action = actionNextArray[0].strip();
+            switch(action) {
+                case "Todo": {
+                    String[] taskArray = nextLine.split(":");
+                    String task = taskArray[1].strip();
+                    Task savedTask = new Todo(task);
+                    if (nextLine.substring(1, 2).equals("X")) {
+                        savedTask.check();
+                    }
+                    tasksList.add(savedTask);
+                    break;
+                }
+                case "Deadline": {
+                    String[] frontTaskArray = nextLine.split(":");
+                    String[] backTaskArray = frontTaskArray[1].split("\\[");
+                    String task = backTaskArray[0].strip();
+                    String[] frontByArray = nextLine.split("\\[");
+                    String[] backByArray = frontByArray[1].split("]");
+                    String by = backByArray[0].strip();
+                    LocalDate storageBy = LocalDate.parse(by, DateTimeFormatter.ofPattern("MMM dd yyyy"));
+                    String outputBy = storageBy.format(DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+                    Task savedTask = new Deadline(task, outputBy);
+                    if (nextLine.substring(1, 2).equals("X")) {
+                        savedTask.check();
+                    }
+                    tasksList.add(savedTask);
+                    break;
```
This one can apply SLAP (Single Level of Abstraction Principle) and change to a method, especially since there are similar code used in other case blocks too.

UnwilledTangent commented on src/main/java/duke/Storage.java:
```diff
@@ -0,0 +1,114 @@
+                    break;
+                }
+                case "Event": {
+                    String[] frontTaskArray = nextLine.split(":");
+                    String[] backTaskArray = frontTaskArray[1].split("\\[");
+                    String task = backTaskArray[0].strip();
+                    String[] frontFromArray = nextLine.substring(3).split("\\[");
+                    String[] backFromArray = frontFromArray[1].split("-");
+                    String from = backFromArray[0].strip();
+                    String[] frontToArray = nextLine.split("-");
+                    String[] backToArray = frontToArray[1].split("]");
+                    String to = backToArray[0].strip();
+                    LocalDate storageFrom = LocalDate.parse(from, DateTimeFormatter.ofPattern("MMM dd yyyy"));
+                    String outputFrom = storageFrom.format(DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+                    LocalDate storageTo = LocalDate.parse(to, DateTimeFormatter.ofPattern("MMM dd yyyy"));
+                    String outputTo = storageTo.format(DateTimeFormatter.ofPattern("yyyy-MM-dd"));
+                    Task savedTask = new Event(task, outputFrom, outputTo);
+                    if (nextLine.substring(1, 2).equals("X")) {
+                        savedTask.check();
+                    }
+                    tasksList.add(savedTask);
+                    break;
```
This one can apply SLAP (Single Level of Abstraction Principle) and change to a method, especially since there are similar code used in other case blocks too.

UnwilledTangent reviewed (commented):
For Week 5 Tutorial

BigheartSmallbrain committed:
Merge branch 'branch-A-CheckStyle'

BigheartSmallbrain committed:
Merge branch 'branch-Level-10'

BigheartSmallbrain committed:
Update buiild.gradle

Change the shadow pluggin mainClassName to duke.Main so
that JavaFX libraries are included when generating a new
JAR file.

BigheartSmallbrain committed:
GUI does not display same message as terminal

GUI display correct respond message for each command

GUI with correct message let user know the app is working fine

Updated DiaglogBox.java and DialogBox.fxml files to support the adjustments

BigheartSmallbrain committed:
Fix GUI for Duke app

BigheartSmallbrain committed:
Merge branch 'branch-Level-10'

BigheartSmallbrain committed:
Add Assertion

BigheartSmallbrain committed:
Update files to improve code quality

BigheartSmallbrain committed:
Merge pull request #2 from BigheartSmallbrain/branch-A-Assertions

Add Assertion

BigheartSmallbrain committed:
Merge pull request #3 from BigheartSmallbrain/branch-A-CodeQuality

Improve code quality

BigheartSmallbrain committed:
Add extension Help

BigheartSmallbrain committed:
Merge pull request #4 from BigheartSmallbrain/branch-C-Help

Add extension Help

BigheartSmallbrain committed:
Add more GUI

BigheartSmallbrain committed:
Add user guide

BigheartSmallbrain committed:
Update Ui.png

BigheartSmallbrain committed:
update jar file to run on any OS
